### PR TITLE
fix: Added bgColor attribute to fix the problem that the background c…

### LINKED ISSUE #42213: 二维码组件设置背景颜色后下载的图片显示不对
### Reproduction link

[Edit on CodeSandbox](https://codesandbox.io/s/xia-zai-er-wei-ma-antd-5-4-7-forked-tg2ggf?file=/demo.tsx)

### Steps to reproduce

直接点击下载

### What is expected?

下载下来的二维码图片颜色是红色的

### What is actually happening?

下载下来的二维码图片颜色是黑色的

| Environment | Info |
| --- | --- |
| antd | 5.4.7 |
| React | 18.0.0 |
| System | macos 13.2 (22D49) |
| Browser | chrome 112.0.5615.137 |

---
我已经解决了这个bug并且提交了个pr.

### PR DESCRIPTION
**Pull request by duriann**
color of the QR code is displayed incorrectly when downloading

[[中文版模板 / Chinese template](https://github.com/ant-design/ant-design/blob/master/.github/PULL_REQUEST_TEMPLATE/pr_cn.md?plain=1)]

### 🤔 This is a ...

- [] New feature
- [x] Bug fix
- [ ] Site / documentation update
- [ ] Demo update
- [ ] Component style update
- [ ] TypeScript definition update
- [ ] Bundle size optimization
- [ ] Performance optimization
- [ ] Enhancement feature
- [ ] Internationalization
- [ ] Refactoring
- [ ] Code style optimization
- [ ] Test Case
- [ ] Branch merge
- [ ] Workflow
- [ ] Other (about what?)

### 🔗 Related issue link

 close https://github.com/ant-design/ant-design/issues/42213
### 📝 Changelog

| Language   | Changelog |
| ---------- | --------- |
| 🇺🇸 English |     add bgColor to set qrcode backgroundColor      |
| 🇨🇳 Chinese |    新增bgColor属性设置二维码背景色      |

### ☑️ Self-Check before Merge

⚠️ Please check all items below before requesting a reviewing. ⚠️

- [x] Doc is updated/provided or not needed
- [x ] Demo is updated/provided or not needed
- [x] TypeScript definition is updated/provided or not needed
- [ ] Changelog is provided or not needed

---

### <samp>🤖 Generated by Copilot at 5a2bffc</samp>

Added a new `bgColor` prop to the QRCode component to enable customizing the background color of the QR code. Updated the component examples, interface, and documentation accordingly.

### <samp>🤖 Generated by Copilot at 5a2bffc</samp>

*  Add `bgColor` prop to QRCode component to customize the background color of the QRCode ([link](https://github.com/ant-design/ant-design/pull/42214/files?diff=unified&w=0#diff-99175e9424e53dc9d08dd9c77c57c2ac26a0aff2f9b4eb0f6cf34d1709ab5662R32), [link](https://github.com/ant-design/ant-design/pull/42214/files?diff=unified&w=0#diff-51a08ab3cfcd9c949cdb0008a7999563efd617f54458e798d427f5fd373c32ddR19))
*  Pass `bgColor` prop to QRCodeGenerator options and useMemo dependency array in `components/qrcode/index.tsx` to render the QRCode with the correct background color ([link](https://github.com/ant-design/ant-design/pull/42214/files?diff=unified&w=0#diff-99175e9424e53dc9d08dd9c77c57c2ac26a0aff2f9b4eb0f6cf34d1709ab5662L50-R55))
*  Document `bgColor` prop in English and Chinese API sections of `components/qrcode/index.en-US.md` and `components/qrcode/index.zh-CN.md` with default value of `transparent` ([link](https://github.com/ant-design/ant-design/pull/42214/files?diff=unified&w=0#diff-d164e5c30509e0d3b0b1fa0d27ebd565deec15bdd561631960505d8d38f17c64R44), [link](https://github.com/ant-design/ant-design/pull/42214/files?diff=unified&w=0#diff-e592d572776f1ad6d667aa605e0bb5732d26d44c6c8da45460cbffea8fbbb400R45))
*  Replace `style` prop by `bgColor` prop in QRCode component examples in `components/qrcode/demo/customColor.tsx` and `components/qrcode/demo/customColor.md` to demonstrate how to use the new prop ([link](https://github.com/ant-design/ant-design/pull/42214/files?diff=unified&w=0#diff-9d6a0d60f6a31ccfec6d76ba34b5c7799a89e4e3c7dc4c2cd924cef4139d2b4bL13-R20), [link](https://github.com/ant-design/ant-design/pull/42214/files?diff=unified&w=0#diff-b94fe0aadfa08ad800f7f5845822ad96953c07cc983a8a018fdd26bc75efa6eaL3-R3))
*  Adjust import order of React and antd components in `components/qrcode/demo/customColor.tsx` to follow convention of importing React first ([link](https://github.com/ant-design/ant-design/pull/42214/files?diff=unified&w=0#diff-9d6a0d60f6a31ccfec6d76ba34b5c7799a89e4e3c7dc4c2cd924cef4139d2b4bL1-R2))

### CODE REVIEW

**Comment by li-jia-nan** [comment]
改到 feature branch 吧，另外加个测试用例

**Comment by MadCcc** [comment]
`bgColor` 也要用到容器上，不然会有白边

**Comment by li-jia-nan** [comment]
rebase 一下，你把 master 的改动带过来了，master 的 commit 不应该出现在这里

**Comment by afc163** [comment]
图片

**Comment by bolosea** [comment]
> 

不会有这个问题 那个容器是用户自己写的div，下载的只是二维码的真实图片内容

**Comment by bolosea** [comment]
> 图片

现在应该好了，第一次提pr不懂操作不好意思= =

**Comment by li-jia-nan** [comment]
同步一下上游分支

**Comment by afc163** [comment]
lint 挂了

**Comment by MadCcc** [comment]
背景是 antd 提供的，也要考虑进去。现在的 demo 和线上的就有区别
image
建议保留线上 demo 的样式，用新 feature 实现。

File: components/qrcode/demo/customColor.tsx [hunk 1]
```diff
@@ -10,12 +10,14 @@ const App: React.FC = () => {
...
-        style={{ marginBottom: 16, backgroundColor: token.colorBgLayout }}
+        style={{ marginBottom: 16 }}
+        bgColor="transparent"
       />
       <QRCode
         value="https://ant.design/"
         color={token.colorInfoText}
-        style={{ marginBottom: 16, backgroundColor: token.colorBgLayout }}
+        style={{ marginBottom: 16 }}
+        bgColor="skyblue"
```
Comment (li-jia-nan) [inline components/qrcode/demo/customColor.tsx:20]: demo 里面建议还是用 token

File: components/qrcode/index.zh-CN.md [hunk 2]
```diff
@@ -42,6 +42,7 @@ group:
 | size | 二维码大小 | number | 160 |
 | iconSize | 二维码中图片的大小 | number | 40 |
 | color | 二维码颜色 | string | `#000` |
+| bgColor | 二维码背景颜色 | string | `transparent` |
```
Comment (li-jia-nan) [inline components/qrcode/index.zh-CN.md:45]: 后面需要加上版本号

File: components/qrcode/index.en-US.md [hunk 3]
```diff
@@ -41,6 +41,7 @@ Used when the text needs to be converted into a QR Code.
 | size | QRCode size | number | 128 |
 | iconSize | include image size | number | 32 |
 | color | QRCode Color | string | `#000` |
+| bgColor | QRCode Background Color | string | `transparent` |
```
Comment (li-jia-nan) [inline components/qrcode/index.en-US.md:44]: 后面需要加上版本号

File: components/qrcode/demo/customColor.tsx (hunk 1 above)
Comment (bolosea) [reply components/qrcode/demo/customColor.tsx:20]: ok

File: components/qrcode/index.en-US.md [hunk 4]
```diff
@@ -34,15 +34,15 @@ Used when the text needs to be converted into a QR Code.
...
+| Property | Description | Type | Default | Version |
+| :-- | :-- | :-- | :-- | :-- |
 | value | scanned text | string | - |
 | icon | include image url (only image link are supported) | string | - |
 | size | QRCode size | number | 128 |
 | iconSize | include image size | number | 32 |
 | color | QRCode Color | string | `#000` |
 | bgColor | QRCode Background Color | string | `transparent` |
-| bordered | Whether has border style | boolean | `true` |
+| bordered | Whether has border style | boolean | `true` | 5.5.0 |
```
Comment (li-jia-nan) [inline components/qrcode/index.en-US.md:45]: 这个版本号是不是加错地方了？

Comment (bolosea) [reply components/qrcode/index.en-US.md]: 你发现的也太快了吧 我刚刚推送完就看到了，重推好了。。

**Comment by li-jia-nan** [comment]
CI 挂了，需要更新一下 snap

File: components/qrcode/demo/customColor.tsx [hunk 5]
```diff
@@ -10,12 +10,14 @@ const App: React.FC = () => {
       <QRCode
         value="https://ant.design/"
         color={token.colorSuccessText}
-        style={{ marginBottom: 16, backgroundColor: token.colorBgLayout }}
+        style={{ marginBottom: 16 }}
+        bgColor="transparent"
```
Comment (li-jia-nan) [inline components/qrcode/demo/customColor.tsx:14]: 这个透明色不用写的，默认就是透明，建议换成 token 里面的颜色，另外这两个 marginBottom 好像可以去掉了

File: components/qrcode/index.tsx [hunk 6]
```diff
@@ -75,7 +76,7 @@ const QRCode: React.FC<QRCodeProps> = (props) => {
   });
 
   return wrapSSR(
-    <div style={{ ...style, width: size, height: size }} className={cls}>
+    <div style={{ ...style, width: size, height: size, backgroundColor: bgColor }} className={cls}>
```
Comment (afc163) [inline components/qrcode/index.tsx:79]: bgColor 为 transparent 时应该不需要给 backgroundColor 。

Comment (li-jia-nan) [reply components/qrcode/index.tsx:79]: transparent 只是默认值，如果用户传进来就不是 transparent 了

**Review by li-jia-nan — APPROVED** [review]

**Pull request merged**